odxtools/additionalaudience.py

```diff
@@ -1,38 +1,26 @@
 # SPDX-License-Identifier: MIT
 from dataclasses import dataclass
-from typing import TYPE_CHECKING, Any, Dict, List, Optional
+from typing import TYPE_CHECKING, Any, Dict, List
 from xml.etree import ElementTree
 
-from .exceptions import odxrequire
+from .element import IdentifiableElement
 from .odxlink import OdxDocFragment, OdxLinkDatabase, OdxLinkId
-from .utils import create_description_from_et
 
 if TYPE_CHECKING:
     from .diaglayer import DiagLayer
 
 
 @dataclass
-class AdditionalAudience:
+class AdditionalAudience(IdentifiableElement):
     """
     Corresponds to ADDITIONAL-AUDIENCE.
     """
 
-    odx_id: OdxLinkId
-    short_name: str
-    long_name: Optional[str]
-    description: Optional[str]
-
     @staticmethod
     def from_et(et_element: ElementTree.Element,
                 doc_frags: List[OdxDocFragment]) -> "AdditionalAudience":
 
-        odx_id = odxrequire(OdxLinkId.from_et(et_element, doc_frags))
-        short_name = odxrequire(et_element.findtext("SHORT-NAME"))
-        long_name = et_element.findtext("LONG-NAME")
-        description = create_description_from_et(et_element.find("DESC"))
-
-        return AdditionalAudience(
-            odx_id=odx_id, short_name=short_name, long_name=long_name, description=description)
+        return AdditionalAudience(**IdentifiableElement.get_kwargs(et_element, doc_frags))
```

Comment:
Problem with `__init_from_et__` is that it does not check if a property was assigned twice, or was not assigned at all

The TypedDict solution gets checked for correctness with mypy

Reply:
fair. The `__init_from_et__` pattern still feels more natural to me, but if the TypedDicts lead to better mypy coverage, let's go that route. That said, I think that There Can Be Only One (TM), so the current users of `__init_from_et__()` should be converted sooner or later.

BTW: how to "stack" these TypedDict objects? (i.e., The one for ComplexComparam would contain the one for BaseComparam which would contain the one for IdentifiableElement)

Reply:
okay: the "attribute missing" problem of the `__init_from_et__` pattern can be solved by calling

```python
def dataclass_is_initialized(dc_obj: Any) -> bool:
    for field in dc_obj.__dataclass_fields__.values():
        if not hasattr(dc_obj, field.name):
            return False

    return True
```

at the end of all `from_et()`  methods. That said, since this would be a runtime check, a static solution would be preferable IMO. (Also, it might still be possible to assign objects of the wrong type to the attributes, but I suppose that mypy will catch that (haven't checked).)

Reply:
@andlaus check the new mechanism in the last commit, works with mypy 1.5